gymnasium/dev_wrappers/__init__.py

```diff
@@ -0,0 +1,5 @@
+"""Root __init__ of the gym dev_wrappers."""
+from typing import Dict, Sequence, TypeVar, Union
+
+ArgType = TypeVar("ArgType")
+FuncArgType = Union[ArgType, Dict[str, "FuncArgType"], Sequence["FuncArgType"]]
```

Comment:
Can you explain what is the purpose of this whole construction? My intuition is that it's way overengineered (or rather, too clever, which makes it harder to maintain), but maybe there's a valid reason for it. But it seems to support arbitrarily nested dicts/sequences of arbitrary objects?

Reply:
Yes, the plan is to support **all** observation / action spaces which includes Dict(Dict(...), ...) like structures. 
Therefore, we need to have functions that can run recursively over spaces, along with argument structures